gateway/src/http_entrypoints.rs

```diff
@@ -298,6 +298,45 @@ struct UpdatePreparationProgress {
     total: u32,
 }
 
+/// Result of reading an SP sensor.
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+    Deserialize,
+    JsonSchema,
+)]
+pub struct SpSensorReading {
+    /// SP-centric timestamp.
+    pub timestamp: u64,
+    /// Value (or error) from the sensor.
+    pub result: SpSensorReadingResult,
+}
+
+/// Single reading (or error) from an SP sensor.
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    PartialOrd,
+    Deserialize,
+    Serialize,
+    JsonSchema,
+)]
+#[serde(tag = "kind", rename_all = "snake_case")]
+pub enum SpSensorReadingResult {
+    Success { value: f32 },
+    DeviceOff,
+    DeviceError,
+    DeviceNotPresent,
+    DeviceUnavailable,
```

Comment:
What is the distinction between these two error types?

Reply:
The comments on these are directly lifted from the [hubris enum defining them](https://github.com/oxidecomputer/hubris/blob/1145c393818d70f9c73670540a7e90c556ef41ae/task/sensor-types/src/lib.rs#L70-L76) :sweat_smile:.

I think it is up to the tasks feeding data into the `sensor` task to decide what these mean. From the `thermal` task, I believe these are:

* `DeviceNotPresent` is sent when a "dynamic input" (e.g., a transceiver) has been removed
* `DeviceUnavailable` is sent when we expect there to be a sensor present, but we haven't gotten any data from it

Reply:
Oh that makes sense, for sensors where we have an out-of-band presence mechanism like the QSFPs. Thanks!